graylog2-server/src/main/java/org/graylog2/indexer/cluster/Cluster.java

```diff
@@ -158,14 +176,37 @@ public boolean isHealthy() {
         }
     }
 
-    public void waitForConnectedAndHealthy(long timeout, TimeUnit unit) throws InterruptedException, TimeoutException {
-        LOG.debug("Waiting until cluster connection comes back and cluster is healthy, checking once per second.");
+    /**
+     * Check if the deflector (write index) health status is not {@link ClusterHealthStatus#RED} and that the
+     * {@link org.graylog2.indexer.Deflector#isUp() deflector is up}.
+     *
+     * @return {@code true} if the deflector is healthy and up, {@code false} otherwise
+     */
+    public boolean isDeflectorHealthy() {
+        try {
+            return deflectorHealth().getStatus() != ClusterHealthStatus.RED && deflector.isUp();
+        } catch (ElasticsearchException e) {
+            LOG.trace("Couldn't determine deflector index health properly", e);
+            return false;
+        }
+    }
+
+    /**
+     * Blocks until the Elasticsearch cluster and current write index is healthy again or the given timeout fires.
+     *
+     * @param timeout the timeout value
+     * @param unit the timeout unit
+     * @throws InterruptedException
+     * @throws TimeoutException
+     */
+    public void waitForConnectedAndDeflectorHealthy(long timeout, TimeUnit unit) throws InterruptedException, TimeoutException {
+        LOG.debug("Waiting until cluster connection comes back and the write index is healthy, checking once per second.");
 
         final CountDownLatch latch = new CountDownLatch(1);
         final ScheduledFuture<?> scheduledFuture = scheduler.scheduleAtFixedRate(() -> {
             try {
-                if (isConnected() && isHealthy()) {
-                    LOG.debug("Cluster is healthy again, unblocking waiting threads.");
+                if (isConnected() && isDeflectorHealthy()) {
+                    LOG.debug("Cluster and write index is healthy again, unblocking waiting threads.");
```

Comment:
I think this should be "or".

I would remove the part about the cluster completely: "Write-active index is healthy again, unblocking waiting threads."